pkg/drainer/drainer.go

```diff
@@ -26,74 +26,97 @@ import (
 	"k8s.io/kubectl/pkg/drain"
 )
 
-var drainHelper *drain.Helper
-var node *corev1.Node
+type Drainer struct {
+	nthConfig   config.Config
+	drainHelper *drain.Helper
+}
 
-//InitDrainer will ensure the drainer has all the resources necessary to complete a drain
-func InitDrainer(nthConfig config.Config) {
-	if drainHelper == nil {
-		drainHelper = getDrainHelper(nthConfig)
-	}
-	node = &corev1.Node{}
-	var err error
-	if !nthConfig.DryRun {
-		node, err = drainHelper.Client.CoreV1().Nodes().Get(nthConfig.NodeName, metav1.GetOptions{})
-		if err != nil {
-			log.Fatalf("Couldn't get node %q: %s\n", nthConfig.NodeName, err.Error())
-		}
-		log.Printf("Successufully retrieved node: %s", node.Name)
+// InitDrainer will ensure the drainer has all the resources necessary to complete a drain.
+// This function must be called before any function in the drainer package.
+func New(nthConfig config.Config) (*Drainer, error) {
+	drainHelper, err := getDrainHelper(nthConfig)
+	if err != nil {
+		log.Println("Unable to construct a drainer because a kubernetes drainHelper could not be built: ", err)
+		return &Drainer{}, err
```

Comment:
best practice would be to return `nil, err` here, not a pointer to an actual Drainer struct.